src/server/src/main/java/io/cassandrareaper/management/MetricsProxy.java

```diff
@@ -17,39 +17,149 @@
 
 package io.cassandrareaper.management;
 
+import io.cassandrareaper.core.DroppedMessages;
 import io.cassandrareaper.core.GenericMetric;
+import io.cassandrareaper.core.JmxStat;
+import io.cassandrareaper.core.MetricsHistogram;
 import io.cassandrareaper.core.Node;
+import io.cassandrareaper.core.ThreadPoolStat;
 import io.cassandrareaper.management.http.HttpCassandraManagementProxy;
 import io.cassandrareaper.management.http.HttpMetricsProxy;
 import io.cassandrareaper.management.jmx.JmxCassandraManagementProxy;
 import io.cassandrareaper.management.jmx.JmxMetricsProxy;
 
 import java.io.IOException;
 import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
 import javax.management.JMException;
 
+import com.google.common.collect.Lists;
+import org.joda.time.DateTime;
+
+
 public interface MetricsProxy {
 
-  static MetricsProxy create(ICassandraManagementProxy proxy) {
+  static MetricsProxy create(ICassandraManagementProxy proxy, Node node) {
     if (proxy == null) {
       throw new RuntimeException("ICassandraManagementProxy is null");
     }
     if (proxy instanceof JmxCassandraManagementProxy) {
-      return JmxMetricsProxy.create((JmxCassandraManagementProxy) proxy);
+      return JmxMetricsProxy.create((JmxCassandraManagementProxy) proxy, node);
     } else if (proxy instanceof HttpCassandraManagementProxy) {
-      return HttpMetricsProxy.create((HttpCassandraManagementProxy) proxy);
+      return HttpMetricsProxy.create((HttpCassandraManagementProxy) proxy, node);
     }
     throw new UnsupportedOperationException("Unknown ICassandraManagementProxy implementation");
   }
 
-  List<GenericMetric> collectTpStats(Node node) throws JMException, IOException;
+  List<GenericMetric> collectTpStats() throws JMException, IOException;
 
-  List<GenericMetric> collectDroppedMessages(Node node) throws JMException, IOException;
+  List<GenericMetric> collectDroppedMessages() throws JMException, IOException;
 
-  List<GenericMetric> collectLatencyMetrics(Node node) throws JMException, IOException;
+  List<GenericMetric> collectLatencyMetrics() throws JMException, IOException;
 
-  List<GenericMetric> collectGenericMetrics(Node node) throws JMException, IOException;
+  List<GenericMetric> collectGenericMetrics() throws JMException, IOException;
 
-  List<GenericMetric> collectPercentRepairedMetrics(Node node, String keyspaceName)
+  List<GenericMetric> collectPercentRepairedMetrics(String keyspaceName)
       throws JMException, IOException;
+
+  static List<GenericMetric> convertToGenericMetrics(Map<String, List<JmxStat>> jmxStats, Node node) {
```

Comment:
That doesn't seem to be possible since the creation of a MetricsProxy instance is done through a static method, which can't initialize a member of the implementation class.